website/addons/wiki/views.py

```diff
@@ -334,6 +350,39 @@ def project_wiki_edit_post(auth, wname, **kwargs):
         ret = {'status': 'success'}
     return ret, http.FOUND, None, redirect_url
 
+@must_be_valid_project  # injects node or project
+@must_have_permission('admin')
+@must_not_be_registration
+@must_have_addon('wiki', 'node')
+def edit_wiki_permissions(node, auth, permissions, **kwargs):
+    wiki_settings = node.get_addon('wiki')
+
+    if not (permissions and wiki_settings):
+        raise HTTPError(http.BAD_REQUEST)
```

Comment:
Return a useful error message to the client by passing `message_long` in the `data` argument.